src/libstrongswan/ipsec/ipsec_types.c

```diff
@@ -150,6 +150,22 @@ bool mark_from_string(const char *value, mark_op_t ops, mark_t *mark)
 	return TRUE;
 }
 
+/*
+ * Described in header
+ */
+uint32_t allocate_unique_mark(void)
+{
+	static refcount_t unique_mark = 0;
```

Comment:
Why define this here and not pass the counter? (Would make it e.g. testable.)

Reply:
Mostly to align it with the other functions herein, namely `allocate_unique_if_ids()`. It makes the use of these functions somewhat simpler by hiding the underlying counter.

I'm fine with either approach, let me know if I shall change it.

Reply:
I see. However, that immediately begs the question whether an approach similar to `allocate_unique_mark()` would also make sense for `allocate_unique_if_ids()`. It's probably similarly unlikely to be a problem as with the marks, but would be consistent.

Reply:
I've updated the changes to use the same mechanic for both mark and interface ID allocation, and handle overflows gracefully in both.

The code duplication is somewhat more obvious now, but I don't see a simple way for unification that is worth the hassle.

Reply:
Thanks, looks good. I'll merge it to master.

> The code duplication is somewhat more obvious now, but I don't see a simple way for unification that is worth the hassle.

Agreed.